batchflow/plot.py

```diff
@@ -0,0 +1,830 @@
+
+        # ylabel
+        keys = ['ylabel'] + text_keys
+        params = self.filter_dict(ax_config, keys, prefix='ylabel_', index=idx)
+        if params:
+            ax.set_ylabel(**params)
+
+        # xticks
+        params = self.filter_dict(ax_config, ['xticks'], prefix='xticks_', index=idx)
+        if 'xticks' in params:
+            params['ticks'] = params.get('ticks', params.pop('xticks'))
+        if params:
+            ax.set_xticks(**params)
+
+        # yticks
+        params = self.filter_dict(ax_config, ['yticks'], prefix='yticks_', index=idx)
+        if 'yticks' in params:
+            params['ticks'] = params.get('ticks', params.pop('yticks'))
+        if params:
+            ax.set_yticks(**params)
+
+        # ticks
+        keys = ['labeltop', 'labelright', 'labelcolor', 'direction']
+        params = self.filter_dict(ax_config, keys, prefix='tick_', index=idx)
+        if params:
+            ax.tick_params(**params)
+
+        # xlim
+        params = self.filter_dict(ax_config, ['xlim'], prefix='xlim_', index=idx)
+        if 'xlim' in params:
+            params['left'] = params.get('left', params.pop('xlim'))
+        if params:
+            ax.set_xlim(**params)
+
+        # ylim
+        params = self.filter_dict(ax_config, ['ylim'], prefix='ylim_', index=idx)
+        if 'ylim' in params:
+            params['bottom'] = params.get('bottom', params.pop('ylim'))
+        if params:
+            ax.set_ylim(**params)
+
+        # colorbar
+        if any(to_list(self.config['colorbar'])):
+            keys = ['colorbar', 'size', 'pad', 'fake', 'ax_image']
+            params = self.filter_dict(ax_config, keys, prefix='colorbar_', index=idx)
+            params['ax_image'] = ax_image
+            # if colorbar is disabled for subplot, add param to plot fake axis instead to keep proportions
+            params['fake'] = not params.pop('colorbar', True)
+            self.add_colorbar(**params)
+
+        # legend
+        keys = ['label', 'size', 'cmap', 'color', 'loc', 'legend', 'ha', 'va']
+        params = self.filter_dict(ax_config, keys, prefix='legend_')
+        params['color'] = params.pop('color', params.pop('cmap', None))
+        params['label'] = params.pop('legend', None) or params.get('label')
+        if params.get('label') is not None:
+            self.add_legend(ax, mode=mode, **params)
+
+        # grid
+        keys = ['grid', 'b', 'which', 'axis']
+        params = self.filter_dict(ax_config, keys, prefix='grid_', index=idx)
+        params['b'] = params.pop('grid', params.pop('b', 'False'))
+        if params:
+            ax.grid(**params)
+
+        facecolor = ax_config.get('facecolor', None)
+        if facecolor is not None:
+            ax.set_facecolor(facecolor)
+
+        ax.set_axisbelow(ax_config.get('set_axisbelow', False))
+
+        if ax_config.get('disable_axes'):
+            ax.set_axis_off()
+        elif not ax.axison:
+            ax.set_axis_on()
+
+    def show(self):
```

Comment:
What about adding `finalize` option/method somewhere to explicitly clear everything?